docs/create_your_own_brick.rst

```diff
@@ -0,0 +1,249 @@
+Create your own brick
+=====================
+
+This tutorial explains how to create a custom brick, which is useful if you
+want to factorize a specific sequence of operations (which can be made of
+bricks themselves) into a single operation so that you can easily reuse it.
+
+The first part of this tutorial lists the requirements and optional components
+that a brick should/can implement while the second part describes the
+construction of a simple toy brick.
+
+This tutorial assumes that you are already familiar with
+:doc:`bricks <bricks_overview>`.
+
+
+Bricks ingredients and recipe
+-----------------------------
+
+All the bricks in blocks inherit directly or indirectly from the
+:class:`.Brick`. However there is already a rich inheritance hierarchy of
+bricks implemented in blocks and thus, you should consider which brick level
+you wish to inherit from. Bear it mind that multiple inheritance is often
+possible and advocated whenever it makes sense.
+
+Here are examples of possible bricks to inherit from:
+
+* :class:`.Sequence`: a sequence of bricks.
+* :class:`.Initializable`: a brick that defines a same initialiation scheme
+  (weights and biases) for all its children.
+* :class:`.Feedforward`: declares an interface for bricks with one input and
+  one output.
+* :class:`.Linear`: a linear transformation with optional bias. Inherits from
+  :class:`.Initializable` and :class:`.Feedforward`.
+* many mores!
+
+Let's say that you want to create a brick from scracth, simply inheriting
+from :class:`.Brick`, then you should consider overwriting the following
+methods (strictly speaking, all these methods are optional):
+
+* :meth:`.Brick.__init__`: you should pass by argument the attributes of your
+  brick. It is also in this method that you should create the potential
+  "children bricks" that belongs to your brick (in that case, you have to put
+  the children bricks into ``self.children``. The initialiazation of the
+  attributes can be lazy as described later in the tutorial.
```

Comment:
I think, you need to mention that the brick can call application methods of other brick only if it is a child brick.